core/vote/vote_manager.go

```diff
@@ -69,6 +73,7 @@ func NewVoteManager(eth Backend, chainconfig *params.ChainConfig, chain *core.Bl
 
 	// Subscribe to chain head event.
 	voteManager.chainHeadSub = voteManager.chain.SubscribeChainHeadEvent(voteManager.chainHeadCh)
+	voteManager.syncVoteSub = voteManager.pool.SubscribeNewVoteEvent(voteManager.syncVoteCh)
 
```

Comment:
need call `Unsubscribe()` when before exit the loop.
both `chainHeadSub` and `syncVoteSub`

Reply:
accepted.

normally, votepool and votemanager will only exit the loop when bsc is shutting down,
so not to unsubscribe() has no bad affect.
but there is really mini chance for votepool and votemanager exiting the loop unnormally.